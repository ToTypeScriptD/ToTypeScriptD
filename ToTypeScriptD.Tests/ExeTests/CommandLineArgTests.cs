using ApprovalTests;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Diagnostics;

namespace ToTypeScriptD.Tests.ExeTests
{
    public class CommandLineArgTests : ExeTestBase
    {
        [Test]
        [Timeout(3000)]
        public void ExeShouldGenerateHelpOnEmptyInput()
        {
            var result = Execute("");

            Assert.IsEmpty(result.StdOut);
            Approvals.Verify(result.StdErr);
        }

        [Test]
        [Timeout(3000)]
        [Ignore("This is hanging on the Console.WriteLine(...)")]
        public void ExeShouldGenerateOutputForMultipleWinmdFiles()
        {
            var result = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Networking.winmd");

            Assert.IsEmpty(result.StdErr);

            result.StdOut.Length.ShouldBeGreaterThan(100);
        }

        [Test]
        [Timeout(3000)]
        [Ignore("This is hanging on the Console.WriteLine(...)")]
        public void ExeDuplicateAssemblyShouldStillOnlyGenerateOne()
        {
            var resultDup = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");
            var resultNonDup = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");

            resultNonDup.ShouldEqual(resultDup);
        }
    }
}
