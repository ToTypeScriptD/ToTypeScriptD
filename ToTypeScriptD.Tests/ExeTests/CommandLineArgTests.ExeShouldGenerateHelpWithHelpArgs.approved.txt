ToTypeScriptD XXX_VERSION_XXX
Copyright (C) 2013 Jason Jarrett
 
Usage: TypeScriptD.exe [--specialTypes] [<file1.winmd> ...<fileN.winmd>]

  -s, --specialTypes    Writes the ToTypeScriptD special types to standard out

  --help                Display this help screen.

